flink/src/main/java/org/apache/iceberg/flink/source/DataIterator.java

```diff
@@ -106,7 +78,9 @@ private void updateCurrentIterator() {
     }
   }
 
-  abstract CloseableIterator<T> openTaskIterator(FileScanTask scanTask) throws IOException;
+  private CloseableIterator<T> openTaskIterator(FileScanTask scanTask) throws IOException {
```

Comment:
The throwing IOException could be removed now because we are throwing CheckedIOException, right ?